runtime/bundles/org.eclipse.core.jobs/src/org/eclipse/core/internal/jobs/InternalJob.java

```diff
@@ -200,7 +200,10 @@ protected boolean cancel() {
 	}
 
 	protected void canceling() {
-		//default implementation does nothing
+		Thread t = getThread();
+		if (t != null) {
+			t.interrupt();
```

Comment:
don't like it because there is no guarantee that that thread is currently running the desired job. The worker threadpool can already be executing another job. The Api is designed such that the Job should activly listen for the monitor.cancel flag. if that is not enough during a IO operation the jjob should spawn a dedicated thread and interrupt that thread when monitor is canceled.

Reply:
The Thread is currently assigned to this job, and the job API is using that thread for different purpose, so I don't understand why there should be an unrelated thread set here?

Reply:
Is it possible that between line 203 and line 205 the job ends and the thread is then used by different job?  I don't see anything that would prevent that from happening...